sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/aio/_data_lake_service_client_async.py

```diff
@@ -86,44 +92,100 @@ class DataLakeServiceClient(AsyncStorageAccountHostsMixin, DataLakeServiceClient
+
         self._client = AzureDataLakeStorageRESTAPI(self.url, base_url=self.url, pipeline=self._pipeline)
-        self._client._config.version = get_api_version(kwargs)  #pylint: disable=protected-access
+        self._client._config.version = get_api_version(kwargs)  # pylint: disable=protected-access
         self._loop = kwargs.get('loop', None)
 
-    async def __aenter__(self):
+    async def __aenter__(self) -> Self:
         await self._blob_service_client.__aenter__()
         return self
 
-    async def __aexit__(self, *args):
+    async def __aexit__(self, *args: Any) -> None:
         await self._blob_service_client.close()
         await super(DataLakeServiceClient, self).__aexit__(*args)
 
-    async def close(self):
-        # type: () -> None
+    async def close(self) -> None:
         """ This method is to close the sockets opened by the client.
         It need not be used when using with a context manager.
         """
         await self.__aexit__()
 
+    def _format_url(self, hostname: str) -> str:
+        """Format the endpoint URL according to hostname.
+
+        :param str hostname: The hostname for the endpoint URL.
+        :returns: The formatted URL
+        :rtype: str
+        """
+        return _format_url(self.scheme, hostname, self._query_str)
+
+    @classmethod
+    def from_connection_string(
+        cls, conn_str: str,
+        credential: Optional[Union[str, Dict[str, str], "AzureNamedKeyCredential", "AzureSasCredential", "TokenCredential"]] = None,  # pylint: disable=line-too-long
+        **kwargs: Any
+    ) -> Self:
+        """
+        Create DataLakeServiceClient from a Connection String.
+
+        :param str conn_str:
+            A connection string to an Azure Storage account.
+        :param credential:
+            The credentials with which to authenticate. This is optional if the
+            account URL already has a SAS token, or the connection string already has shared
+            access key values. The value can be a SAS token string,
+            an instance of a AzureSasCredential from azure.core.credentials, an account shared access
+            key, or an instance of a TokenCredentials class from azure.identity.
+            Credentials provided here will take precedence over those in the connection string.
+        :type credential:
+            ~azure.core.credentials.AzureNamedKeyCredential or
+            ~azure.core.credentials.AzureSasCredential or
+            ~azure.core.credentials.TokenCredential or
+            str or Dict[str, str] or None
+        :keyword str audience: The audience to use when requesting tokens for Azure Active Directory
+            authentication. Only has an effect when credential is of type TokenCredential. The value could be
+            https://storage.azure.com/ (default) or https://<account>.blob.core.windows.net.
+        :returns: A DataLakeServiceClient.
+        :rtype: ~azure.storage.filedatalake.DataLakeServiceClient
```

Comment:
Do we need a sample for async datalake service client here? There's an identical one for the sync version, how about linking that one instead? Or we could leave as is :)

Reply:
After looking at what we did in Blob:
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py#L175
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-blob/azure/storage/blob/aio/_blob_service_client_async.py#L179

And Queue:
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-queue/azure/storage/queue/_queue_service_client.py#L149
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-queue/azure/storage/queue/aio/_queue_service_client_async.py#L141

We just re-used the sync sample. I think this is fine and we definitely shouldn't just omit the sample just because we don't have a specific "async" example (which would just involve importing the async equiv. and maybe an `await` here and there 😆 

tl;dr add the sample and it's fine to reuse the sync sample

Reply:
Personally, I don't really think we need the sample if we don't have one already. This is a pretty easy thing to use. But if you want to add one to match other packages, I won't stop you :D

Reply:
Sure that sounds good!

Reply:
Mainly just like our packages to be as symmetrical as possible for a good UX.

Reply:
The sample already exists, just gotta add to the docs after the decoupling work!